pkg/morph/client/notary.go

```diff
@@ -203,11 +205,10 @@ func (c *Client) UpdateNeoFSAlphabetList(list keys.PublicKeys) error {
 // blockchain. Fallback tx is a `RET`. If Notary support is not enabled
 // it fallbacks to a simple `Invoke()`.
 //
-// This function must be invoked after `EnableNotarySupport()` otherwise it
-// throws panic.
+// This function must be invoked with notary enabled otherwise it throws panic.
 func (c *Client) NotaryInvoke(contract util.Uint160, fee fixedn.Fixed8, method string, args ...interface{}) error {
 	if c.notary == nil {
-		return c.Invoke(contract, fee, method, args...)
+		panic(notaryNotEnabledPanicMsg)
```

Comment:
Not sure about that. Or, at least, we need to make more changes.

Inner ring, depending on config, enables or disables notary in the client. Static clients, which are wrappers, have `TryNotary` option. If option enabled, then static client uses `NotaryInvoke()`, otherwise `Invoke()`. `NotaryInvoke()`, on the other hand, can fallback to `Notary()` if notary was not enabled (you added it in https://github.com/nspcc-dev/neofs-node/commit/b5cda8cd4180db5e6d13b83fe8813923b78e020e).

Now:
1. all static clients in Inner Ring have `TryNotary` option,
2. if Inner Ring config disables notary, then `c.notary == nil`,
3. in this case Inner Ring expects to fallback to `Invoke()`.

(3) is not happening with this change. So you either need to return `Invoke()` fallback or reconsider role of `TryNotary()` flag in static client.